cpp/test/test_supercapacitor_inspector.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2016, the Cap authors.
+ *
+ * This file is subject to the Modified BSD License and may not be distributed
+ * without copyright and license information. Please refer to the file LICENSE
+ * for the text and further information on this license.
+ */
+
+#define BOOST_TEST_MODULE TestSupercapacitorInsepector
+
+#include "main.cc"
+
+#include <cap/energy_storage_device.h>
+#include <cap/supercapacitor.h>
+#include <boost/test/unit_test.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/info_parser.hpp>
+#include <boost/filesystem.hpp>
+
+// Check that the inspector can be used to ouput a distributed mesh
+
+namespace cap
+{
+
+BOOST_AUTO_TEST_CASE(test_supercapacitor_inspector)
+{
+  boost::property_tree::ptree device_database;
+  boost::property_tree::info_parser::read_info("super_capacitor.info",
+                                               device_database);
+
+  std::shared_ptr<cap::EnergyStorageDevice> device =
+      cap::EnergyStorageDevice::build(device_database,
+                                      boost::mpi::communicator());
+
+  cap::SuperCapacitorInspector<2> supercap_inspector;
+  supercap_inspector.inspect(device.get());
+
+  // Check that the files exist
+  if (boost::mpi::communicator().rank() == 0)
+  {
+    BOOST_TEST(boost::filesystem::exists("solution-0000.0000.vtu"));
+    BOOST_TEST(boost::filesystem::exists("solution-0000.0001.vtu"));
+    BOOST_TEST(boost::filesystem::exists("solution-0000.pvtu"));
+  }
```

Comment:
want to get the size and check that the ``.vtu`` file exist for all ranks or add a comment that this is meant to run with two procs?